### PR TITLE
Better DOI term snippet, 3.7+ testing docs

### PR DESCRIPTION
**Pull request by jamesbraza**
- Better DOI searching snippet
    - Improves docstring present
    - Adds `try-except` block
- Documents 3.7+ is tested

### CODE REVIEW

**Comment by jamesbraza** [comment]
Test failures appear to be unrelated to this `README` change

File: README.md [hunk 1]
```diff
@@ -283,27 +283,31 @@ result = entrez_api.search(
     database='pubmed',
     max_results=1
 )
-result.data['esearchresult']['idlist']
+try:
+    print(result.data['esearchresult']['idlist'])
+except KeyError as exc:
+    raise ValueError(
+        f"Unexpected response data blob {result.data}."
+    ) from exc
```
Comment (krassowski) [inline README.md]: I would skip error handling from the examples in the readme. They are meant to be illustrative, not comprehensive.

Comment (jamesbraza) [reply README.md]: Good point, thanks!  Will implement that shortly

Comment (jamesbraza) [reply README.md]: Alright, done!

**Review by krassowski — APPROVED** [review]
Thanks!

**Pull request merged**